types/2020-03-02/Checkout/Sessions.d.ts

```diff
@@ -667,7 +667,7 @@ declare module 'stripe' {
             /**
              * The ID of the product that this price will belong to.
              */
-            product: string;
+            product?: string;
```

Comment:
yeah optional in the spec is finicky and it's really annoying but it's because they are preparing some beta where this is optional because you can pass `product_data` instead